spec/puppet-server-lib/puppet/jvm/ast_compiler_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper'
+
+require 'puppet_pal'
+
+require 'puppet/server/ast_compiler'
+
+def generate_ast(code)
+  Puppet::Pal.in_tmp_environment("ast") do |pal|
+    pal.with_catalog_compiler do |compiler|
+      ast = compiler.parse_string(code)
+      Puppet::Pops::Serialization::ToDataConverter.convert(ast,
+                                                           rich_data: true,
+                                                           symbol_to_string: true)
+    end
+  end
+end
+
+def request(ast)
```

Comment:
Do we ever call `generate_ast` w/o immediately calling `request` on the result?

Reply:
No, and probably don't have to. I can combine this.